source/extensions/filters/listener/proxy_protocol/proxy_protocol.cc

```diff
@@ -533,6 +534,32 @@ bool Filter::parseTlvs(const uint8_t* buf, size_t len) {
     absl::string_view tlv_value(reinterpret_cast<char const*>(buf + idx), tlv_value_length);
     auto key_value_pair = config_->isTlvTypeNeeded(tlv_type);
     if (nullptr != key_value_pair) {
+      if (Runtime::runtimeFeatureEnabled(
+              "envoy.reloadable_features.use_typed_metadata_in_proxy_protocol_listener")) {
+        auto& typed_filter_metadata = (*cb_->dynamicMetadata().mutable_typed_filter_metadata());
+        std::string metadata_key = key_value_pair->metadata_namespace().empty()
+                                       ? "envoy.filters.listener.proxy_protocol"
+                                       : key_value_pair->metadata_namespace();
```

Comment:
I think this is identical to the old path; remove copy/paste and share it between the two paths.